gc_registry/account/validation.py

```diff
@@ -0,0 +1,71 @@
+from fastapi import HTTPException
+from sqlmodel import Session, select
+
+from gc_registry.account.models import Account, AccountWhitelist
+from gc_registry.user.models import User
+
+
+def validate_account(account, read_session):
+    # Make sure operations cannot be performed on deleted accounts
+    if account.is_deleted:
+        raise HTTPException(status_code=400, detail="Cannot update deleted accounts.")
+
+    # Account names must be unique
+    account_exists_query = (
+        select(Account).filter(Account.account_name == account.account_name).exists()
+    )
+    account_exists = read_session.execute(select(account_exists_query)).scalar()
+
+    if account_exists:
+        raise HTTPException(
+            status_code=400, detail="Account name already exists in the database."
+        )
+
+    # All user_ids linked to the account must exist in the database
+    user_ids_in_db = (
+        read_session.query(User.id).filter(User.id.in_(account.user_ids)).all()
+    )
+    if len(user_ids_in_db) != len(account.user_ids):
```

Comment:
Would this check be better a `set` operation to check membership is the same? 

e.g. `set(user_ids_in_db) != set(account.user_ids)`

Reply:
Aye good catch